### DISCUSSION TITLE
azure-pipeline-variables/

### DISCUSSION THREAD

**Discussion by giscus** [colindembovsky/colindembovsky.github.io#9] [Giscussions]
# azure-pipeline-variables/

In this post I take a deep dive into Azure Pipeline variables.

https://colinsalmcorner.com/azure-pipeline-variables/

**Comment by philippfrenzel** [comment]
Hi Colin,
1st thanks for writing the post ;) 2nd can you add another example on how to pass variables to a stage template? 
Thanks a lot and cheers Philipp

**Comment by brunobva** [comment]
Hi Colin, 
Same as Philip, 1st thanks for writing this post. But I am trying to import some variables (all) from my Variable Group.
I've already get the list of all variables, but now, I need to create that, but I can't found (until now) a way to do that. Could please give me a tip?

Really Appreciate your attention and consideration on it!

Thanks
Bruno Moraes

**Comment by adriancuadrado** [comment]
Congratulations for having part of your content plagiarized by Microsoft:
https://learn.microsoft.com/en-us/training/modules/integrate-azure-pipelines/2-describe-anatomy-of-pipeline#variables

> ## Variables
> It would be tough to achieve any sophistication in your pipelines without variables. Though this classification is partly mine, several types of variables exist, and pipelines don't distinguish between these types. However, I've found it helpful to categorize > pipeline variables to help teams understand nuances when dealing with them.
> 
> Every variable is a key: value pair. The key is the variable's name, and it has a value.
> 
> To dereference a variable, wrap the key in $(). Let's consider this example:
> 
> ```yaml
> variables:
>   name: John
> steps:
> 
> - script: echo "Hello, $(name)!"
> ```
> 
> It will write Hello, John! To the log.

**Comment by DotNetWorker** [comment]
To make the variable group available in your pipeline using the UI navigate to:
Pipelines>Your Pipeline> Variables>Variables groups>"Link variable group" 

Use this to link your variable group to your pipeline, then you can pick up the variables as you would if they were defined directly in the pipeline (you also get to specify when linking what kind of build it is for)
